pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1810,18 +1800,23 @@ func (configgen *ConfigGeneratorImpl) onVirtualOutboundListener(
 		FilterChains: make([]plugin.FilterChain, len(ipTablesListener.FilterChains)),
 	}
 
+	// Set the protocol for each filter chain
+	for index := range mutable.FilterChains {
+		mutable.FilterChains[index].ListenerProtocol = plugin.ListenerProtocolTCP
+	}
 	for _, p := range configgen.Plugins {
 		if err := p.OnVirtualListener(pluginParams, mutable); err != nil {
 			log.Warn(err.Error())
 		}
 	}
-	if len(mutable.FilterChains) > 0 && len(mutable.FilterChains[0].TCP) > 0 {
-		filters := append([]*listener.Filter{}, mutable.FilterChains[0].TCP...)
-		filters = append(filters, fallbackFilter)
-
-		// Replace the final filter chain with the new chain that has had plugins applied
-		initialFilterChain = append(initialFilterChain, &listener.FilterChain{Filters: filters})
-		ipTablesListener.FilterChains = initialFilterChain
+	// This logic is same as the one in buildCompleteFilterChain
+	for index := range ipTablesListener.FilterChains {
+		if len(mutable.FilterChains[index].TCP) > 0 {
```

Comment:
IIUC, this will also add Mixer v1 filter for the filter chain that matches pod ip/32 address to prevent infinite loops.

Reply:
don't you want to get telemetry on that traffic before it is being dropped?

Reply:
don't you want to get telemetry on that traffic before it is being dropped?

Reply:
I don't think there is any harm in adding it, I was just pointing out the delta.